test/parallel/test-buffer-writedouble.js

```diff
@@ -67,14 +67,18 @@ assert.strictEqual(buffer.readDoubleLE(8), -Infinity);
 buffer.writeDoubleBE(NaN, 0);
 buffer.writeDoubleLE(NaN, 8);
 
-assert.ok(buffer.equals(new Uint8Array([
-  0x7F, 0xF8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xF8, 0x7F
 // Mips processors use a different NaN.
-])) || buffer.equals(new Uint8Array([
-  0x7F, 0xF7, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
-  0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xF7, 0x7F
-])));
+if (process.arch === 'mips') {
```

Comment:
According to the docs for [`process.arch`](https://github.com/nodejs/node/blob/0cca27b3ffb9fade34aac0ee77d020fb7cac0879/doc/api/process.md#processarch) :

> The current possible values are: `'arm'`, `'arm64'`, `'ia32'`, `'mips'`, `'mipsel'`, `'ppc'`, `'ppc64'`, `'s390'`, `'s390x'`, `'x32'`, and `'x64'`.

so we probably need to be checking for `mipsel` as well. Maybe `mips64el` too. Perhaps @kapouer or anyone else with access to mips can check?

Reply:
Hm, actually, it seems like we either have to use the first approach or we can just check one byte.

https://sourceware.org/binutils/docs/as/MIPS-NaN-Encodings.html

Reply:
If we go back to the first approach, maybe add a comment `Allow both quiet and signalling NaNs`?

Reply:
I have not built nodejs 10 on mipsel yet, but nodejs 8.11 gives the same result as mips on mipsel.